### PR TITLE
WIP: add architecture specific includes for makefiles

### PR DESCRIPTION
**Pull request by rhooper**
This is a work-in-progress to add dependency setup and validation on a per-platform basis.

### CODE REVIEW

File: mk/platform-Darwin.mk [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+THe install-deps target can be used to install dependencies)
+endif
+
+install-deps::
+	@if [[ ! -x "$$(which brew)" ]]; then \
+		echo "Homebrew must be installed; See https://brew.sh/"; \
+		exit 1; \
+  	fi
+	brew install -q git python3 gettext uncrustify rust
+	brew install -q gcc-arm-embedded
```
Comment (jepler) [inline mk/platform-Darwin.mk:37]: note that we request use of a specific version of the arm embedded toolchain, currently 10-2020-q4. Is it possible to request this version in particular via brew? This is hidden down in our github actions:
```
    - uses: carlosperate/arm-none-eabi-gcc-action@v1
      with:
        release: '10-2020-q4'
```

**Review by jepler — COMMENTED** [review]
Personally, I don't like using scripts like these to install software on my computer, since I have Ideas about how things should be. But, this is potentially helpful to some folks! thanks for starting a ball rolling

File: mk/platform-Darwin.mk (hunk 1 above)
Comment (rhooper) [reply mk/platform-Darwin.mk:37]: This was a dual-purpose PR in the end:
- Introduce the idea of platform-specific make includes/settings (eg MacOS - gmake, but also other things like the LD_LIBRARY_PATH/INCLUDES for things like gettext)

- Introduce platform-specific checks/dependency helpers

I'll look into the eabi version.

**Comment by tannewt** [comment]
I'm going to close this because the [Building CircuitPython](https://learn.adafruit.com/building-circuitpython) learn guide is the de facto place for dependency install instructions. Thanks for the PR anyway!

**Pull request closed without merging**